x/oracle/oracle_test.go

```diff
@@ -0,0 +1,282 @@
+package oracle
+
+import (
+	"bytes"
+	"encoding/json"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	abci "github.com/tendermint/abci/types"
+	dbm "github.com/tendermint/tmlibs/db"
+
+	"github.com/tendermint/go-crypto"
+
+	"github.com/cosmos/cosmos-sdk/store"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/wire"
+)
+
+func defaultContext(keys ...sdk.StoreKey) sdk.Context {
+	db := dbm.NewMemDB()
+	cms := store.NewCommitMultiStore(db)
+	for _, key := range keys {
+		cms.MountStoreWithDB(key, sdk.StoreTypeIAVL, db)
+	}
+	cms.LoadLatestVersion()
+	ctx := sdk.NewContext(cms, abci.Header{}, false, nil, nil)
+	return ctx
+}
+
+type validator struct {
+	Address sdk.Address
+	Power   sdk.Rat
+}
+
+func (v validator) GetStatus() sdk.BondStatus {
+	return sdk.Bonded
+}
+
+func (v validator) GetOwner() sdk.Address {
+	return v.Address
+}
+
+func (v validator) GetPubKey() crypto.PubKey {
+	return nil
+}
+
+func (v validator) GetPower() sdk.Rat {
+	return v.Power
+}
+
+func (v validator) GetBondHeight() int64 {
+	return 0
+}
+
+type validatorSet struct {
+	Validators []validator
+}
```

Comment:
why are we reimplementing validator and validatorset when we could just import validator from staking? One less thing to maintain here as the validator set keeper from `types` changes

Reply:
One of the intentions was providing an example how to build an user-defined validatorset. Also, it was hard to update the validatorset manually using Stake.

Reply:
Meh - I think there is better places to have democode to than in this test - if you want a super simple validator set for testing - let's move this to `tests/` as common test suite functionality

Reply:
`mock/` also looks a nice place to put it in.